### ISSUE TITLE
Transition to Rust 2018 and code clean ups

### ISSUE THREAD

**Issue by topecongiro** [rust-lang/rustfmt#3315]
Please help transitioning rustfmt code base to Rust 2018! You can follow the guide below and read [the edition book](https://rust-lang-nursery.github.io/edition-guide/editions/transitioning-an-existing-project-to-a-new-edition.html) for more details.

1. Run `cargo +nightly fix --edition`
2. Sort import groups based on the following order:
    1. standard libraries (those starts with `use std::`)
    2. third-party libraries
    3. local libraries (those start with `use crate::`)
3. Run `cargo fmt`. You may need to use the one built from `master` instead of the one installed via `rustup`
4. Run `cargo build && cargo test` and make sure that the all tests pass!

**Comment by scampi** [comment]
@topecongiro shouldn't we put `edition = "2018"` in `Cargo.toml` ?

**Comment by topecongiro** [comment]
Yup :crying_cat_face: . Created https://github.com/rust-lang/rustfmt/pull/3327.
